File: 2350-find-closest-number-to-zero/2350-find-closest-number-to-zero.go
func findClosestNumber(nums []int) int {
    closest_num := nums[0]
    lowest_dist := abs(nums[0])
    for _, num := range(nums){
        if dist := abs(num); dist < lowest_dist{
            closest_num = num
            lowest_dist = dist
        } else if abs(num) == lowest_dist && num > closest_num{
            closest_num = num
        }
    }
    return closest_num
}

func abs(x int) int{
    if x < 0{
        return -x
    }
    return x
}

File: 2350-find-closest-number-to-zero/2350-find-closest-number-to-zero.py
class Solution:
    def findClosestNumber(self, nums: List[int]) -> int:
        lowest_dist = float('inf')
        closest_num = None
        for num in nums:
            if abs(num) < lowest_dist:
                closest_num = num
                lowest_dist = abs(num)

            elif abs(num) == lowest_dist:
                closest_num = max(closest_num, num)

        return closest_num

        